### PR TITLE
Add flexible voting tests

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description

- Add flexible voting tests

### CODE REVIEW

**Comment by alexkeating** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/6" target="_blank">Graphite</a> 
    * **PR #7** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/7" target="_blank">Graphite</a> 
      * **PR #9** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/9" target="_blank">Graphite</a> 
        * **PR #11** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/11" target="_blank">Graphite</a> 
          * **PR #12** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/12" target="_blank">Graphite</a> 
            * **PR #13** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/13" target="_blank">Graphite</a> 
              * **PR #14** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/14" target="_blank">Graphite</a>  👈
                * **PR #15** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/15" target="_blank">Graphite</a> 
                  * **PR #17** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/17" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/14?utm_source=stack-comment).

**Review by apbendi — APPROVED** [review]
Overall looking good @alexkeating, this one probably good to go once its parents are merged.

File: src/PooltogetherGovernor.sol
Comment (apbendi) [inline src/PooltogetherGovernor.sol:83]: *arbitrary comment location*

@alexkeating the latest commit is currently failing to build with the following error after `forge clean && forge build`

```
[⠊] Compiling...
[⠊] Compiling 65 files with 0.8.18
[⠒] Solc 0.8.18 finished in 939.15ms
Error:
Compiler run failed:
Error (7576): Undeclared identifier.
   --> test/PooltogetherGovernor.t.sol:929:15:
    |
929 |       address(usdcToken), usdcToken.balanceOf(TIMELOCK), makeAddr("receiver for FlexVoting tests")
    |               ^^^^^^^^^

Error (7576): Undeclared identifier.
   --> test/PooltogetherGovernor.t.sol:929:27:
    |
929 |       address(usdcToken), usdcToken.balanceOf(TIMELOCK), makeAddr("receiver for FlexVoting tests")
    |                           ^^^^^^^^^

Comment (alexkeating) [reply src/PooltogetherGovernor.sol:83]: https://github.com/ScopeLift/pooltogether-governor-upgrade/pull/14/commits/1bddeb65663844a2a17d1c97261db4fdba04c6e1 Ah fixed

**Pull request merged**
